server/services/overviewPageService.ts

```diff
@@ -607,20 +622,13 @@ export default class OverviewPageService {
     return []
   }
 
-  private getNeurodiversitySupportStatus(
-    prisonerData: Prisoner,
-    learnerNeurodivergence: LearnerNeurodivergence[],
-  ): Status[] {
-    return (
-      (learnerNeurodivergence?.length &&
-        neurodiversityEnabled(prisonerData.prisonId) && [
-          {
-            label: 'Support needed',
-            subText: 'Has neurodiversity needs',
-          },
-        ]) ||
-      []
-    )
+  private getNeurodiversitySupportStatus(learnerNeurodivergence: Result<LearnerNeurodivergence[]>): Status[] {
+    const supportNeededStatus = { label: 'Support needed', subText: 'Has neurodiversity needs' }
+    const supportNeededErrorStatus = { label: 'Support needs unavailable', subText: 'Try again later', error: true }
+    return learnerNeurodivergence.handle({
```

Comment:
An example of the `Result.handle` function - hopefully it makes it easier to determine what to do in success / failure cases and the typing forces you to handle both cases.